httpserver/bin/note_server.dart

```diff
@@ -6,67 +6,85 @@
 // Use note_taker.html to run the client.
 
 import 'dart:io';
-import 'dart:convert' show JSON;
+import 'dart:async';
+import 'dart:convert' show UTF8, JSON;
 
 int count = 0;
 
-void main() {
+main() async {
   // One note per line.
-  List<String> lines = new File('notes.txt').readAsLinesSync();
-  count = lines.length;
+  try {
+    List<String> lines = new File('notes.txt').readAsLinesSync();
+    count = lines.length;
 
-  HttpServer.bind(InternetAddress.LOOPBACK_IP_V4, 4042)
-      .then(listenForRequests)
-      .catchError((e) => print('hello: ${e.toString()}'));
+    var server =
+        await HttpServer.bind(InternetAddress.LOOPBACK_IP_V4, 4042);
+    print('Listening for requests on 4042.');
+    await listenForRequests(server);
+  } on FileSystemException {
+    print('Could not open notes.txt');
+  } catch (e) {
+    print('hello: ${e.toString()}');
+  }
 }
 
-listenForRequests(_server) {
-  _server.listen((HttpRequest request) {
-    switch (request.method) {
-      case 'POST':
-        handlePost(request);
-        break;
-      case 'OPTION':
-        handleOptions(request);
-        break;
-      default:
-        defaultHandler(request);
-        break;
+Future listenForRequests(HttpServer requests) async {
```

Comment:
This function has a return type of `Future` where other async functions have no return type. Is there a system to that? (like, are you expected to wait for the result)?